### PR TITLE
Make `cholmod` vector solve return vector

### PR DESCRIPTION
**Pull request by dkarrasch**
Fixes https://github.com/JuliaLang/julia/issues/48095.

### CODE REVIEW

File: src/solvers/cholmod.jl [hunk 1]
```diff
@@ -1526,7 +1544,10 @@ end
 function (\)(L::FactorComponent, B::Matrix)
     Matrix(L\Dense(B))
 end
-function (\)(L::FactorComponent, B::SparseVecOrMat)
+function (\)(L::FactorComponent, B::SparseVector)
+    sparsevec(L\Sparse(B))
```
Comment (stevengj) [inline src/solvers/cholmod.jl:1548]: I'm a little confuse about why we are returning a sparse vector — isn't the result generally dense?

File: src/solvers/cholmod.jl [hunk 2]
```diff
@@ -891,10 +903,16 @@ function Hermitian{Tv,SparseMatrixCSC{Tv,SuiteSparse_long}}(A::Sparse{Tv}) where
     ishermitian(A) || throw(ArgumentError("matrix is not Hermitian"))
     args = _extract_args(s, Tv)
     s.sorted == 0 && _sort_buffers!(args...)
-    Hermitian(_trim_nz_builder!(args...), s.stype > 0 ? :U : :L)
+    Hermitian(SparseMatrixCSC(_trim_nz_builder!(args...)...), s.stype > 0 ? :U : :L)
 end
 convert(T::Type{Hermitian{Tv,SparseMatrixCSC{Tv,SuiteSparse_long}}}, A::Sparse{Tv}) where {Tv<:VTypes} = T(A)
 
+function sparsevec(A::Sparse{Float64})
```
Comment (stevengj) [inline src/solvers/cholmod.jl:910]: Why not `Sparse{Tv}`?

File: src/solvers/cholmod.jl (hunk 1 above)
Comment (dkarrasch) [reply src/solvers/cholmod.jl:1548]: I was confused about that myself, but note that previously we returned a `SparseMatrixCSC`. This change only changes the dimension of the returned array. But, of course, we may as well change to dense return arrays. The actual type of the return value doesn't seem to be tested, so I don't know what "promises" we made.

File: src/solvers/cholmod.jl (hunk 2 above)
Comment (dkarrasch) [reply src/solvers/cholmod.jl:910]: True. I was following the pattern below, but in any case the complex version is missing, and also this function by itself need not be restricted.

File: src/solvers/cholmod.jl (hunk 1 above)
Comment (stevengj) [reply src/solvers/cholmod.jl:1548]: I guess we might as well stay consistent here.

See also https://discourse.julialang.org/t/sparse-solve-with-sparse-rhs/81595/3?u=stevengj

**Comment by stevengj** [comment]
LGTM.

**Comment by ViralBShah** [comment]
Merge?

**Pull request merged**